### PR TITLE
RMIO-70 Map IOException to an IOException with a message instead of t…

### PR DESCRIPTION
**Pull request by dukorama**
https://re-motion.atlassian.net/browse/RMIO-70

### CODE REVIEW

File: Archive.Zip/ZipFileBuilder.cs [hunk 1]
```diff
@@ -128,7 +128,7 @@ private void AddFileToZipFile (IFileInfo fileInfo, ZipOutputStream zipOutputStre
         }
         catch (IOException ex)
         {
-          throw new AbortException (
+          throw new IOException (
               string.Format ("Error while copying the data from the file '{0}' to the archive.", fileInfo.FullName),
               ex);
```
Comment (iam-mholle) [inline Archive.Zip/ZipFileBuilder.cs:133]: Does it make sense to include the original `IOException` as InnerException to an `IOException`?

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 2]
```diff
@@ -230,7 +230,7 @@ public void Build_IOExceptionDuringCopyingOccurs_ThrowsAbortExceptionWithIOExcep
               {
               }
             },
-            Throws.InstanceOf<AbortException>()
+            Throws.InstanceOf<IOException>()
```
Comment (iam-mholle) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:233]: The name of the test should be adapted as well.

**Review by iam-mholle — CHANGES_REQUESTED** [review]

File: Archive.Zip/ZipFileBuilder.cs (hunk 1 above)
Comment (dukorama) [reply Archive.Zip/ZipFileBuilder.cs:133]: Yes, to keep the stacktrace of the original exception.

**Review by iam-mholle — APPROVED** [review]
🥳

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**